### PR TITLE
Drop `FieldType` enum

### PR DESCRIPTION
**Pull request by wslulciuc**
This PR removes the `FieldType` enum to allow for any string provided by the caller to be collected. This avoids maintaining an enum of all possible field types. The field type value can be normalized when displayed in the UI (i.e. `varchar(64)` will be normalized to `VARCHAR(64)`), but in the database, the raw value will be persisted or `null` if the field type is not provided (or known). Fixes https://github.com/MarquezProject/marquez/issues/1560

Note, I've opened an issue to discuss normalization of field types https://github.com/MarquezProject/marquez/issues/1562

### CODE REVIEW

File: api/src/main/java/marquez/db/OpenLineageDao.java [hunk 1]
```diff
@@ -537,7 +537,11 @@ default UUID version(
                 fields == null
                     ? ImmutableList.of()
                     : fields.stream()
-                        .map(field -> versionField(field.getName(), field.getType()))
+                        .map(
+                            field ->
+                                versionField(
+                                    field.getName(),
+                                    Utils.toUpperCaseIfPresent(field.getType().toUpperCase())))
```
Comment (OleksandrDvornik) [inline api/src/main/java/marquez/db/OpenLineageDao.java:544]: I guess it should be `Utils.toUpperCaseIfPresent(field.getType())))`

File: api/src/main/java/marquez/db/DatasetVersionDao.java [hunk 2]
```diff
@@ -103,7 +102,7 @@ default DatasetVersionRow upsertDatasetVersion(
               UUID.randomUUID(),
               now,
               field.getName().getValue(),
-              field.getType().name(),
+              Utils.toUpperCaseIfPresent(field.getType()),
```
Comment (OleksandrDvornik) [inline api/src/main/java/marquez/db/DatasetVersionDao.java:105]: proposal, maybe we can add this Utils.toUpperCaseIfPresent() logic to setter? Cause in all places we should do that even on getters, but if we apply that on object creation, we will be sure that it's uppercased

File: api/src/main/java/marquez/common/models/Field.java [hunk 3]
```diff
@@ -36,7 +36,7 @@
   @Getter
   private final FieldName name;
 
-  @Getter private final FieldType type;
+  @Nullable @Getter private final String type;
```
Comment (OleksandrDvornik) [inline api/src/main/java/marquez/common/models/Field.java:39]: should we apply the same for SchemaField to keep them in sync?

**Review by OleksandrDvornik — COMMENTED** [review]
Please take a look on my comments

File: api/src/main/java/marquez/common/models/Field.java (hunk 3 above)
Comment (wslulciuc) [reply api/src/main/java/marquez/common/models/Field.java:39]: We'll eventually want to use the classes published by OpenLineage, but yeah to keep the models in sync I can also update the class `SchemaField`.

**Review by julienledem — APPROVED** [review]
Looks good to me. 
Why did we need to change the python tests? The enum was uppercase before. I'm assuming all types had to be uppercase?

**Comment by wslulciuc** [comment]
> Why did we need to change the python tests? The enum was uppercase before. I'm assuming all types had to be uppercase?

Yeah, when I made the field type optional, and made sure the python client didn't normalize the value by calling `.upper()` (based on https://github.com/MarquezProject/marquez/issues/1562), I had to change the field type casing in the test. We also no longer need to check that an exception is thrown when the field type is not present.

**Pull request merged**